src/gt4py/stencil_builder.py

```diff
@@ -52,9 +52,9 @@ def __init__(
         self.options = options or BuildOptions(  # type: ignore
             **self.default_options_dict(definition_func)
         )
-        self.backend: "BackendType" = (
-            backend(self) if backend else gt4py.backend.from_name("debug")(self)
-        )
+        if not backend_name:
+            backend_name = "debug"
```

Comment:
Not really a suggestion but a comment: there is an alternative and concise way to write this kind of checks:
```suggestion
        backend_name = backend_name or "debug"
```